spanner/ddl/ast.go

```diff
@@ -232,14 +234,16 @@ func (k Foreignkey) PrintForeignKey(c Config) string {
 // CreateTable encodes the following DDL definition:
 //     create_table: CREATE TABLE table_name ([column_def, ...] ) primary_key [, cluster]
 type CreateTable struct {
-	Name     string
-	ColNames []string             // Provides names and order of columns
-	ColDefs  map[string]ColumnDef // Provides definition of columns (a map for simpler/faster lookup during type processing)
-	Pks      []IndexKey
-	Fks      []Foreignkey
-	Indexes  []CreateIndex
-	Parent   string //if not empty, this table will be interleaved
-	Comment  string
+	Name         string
+	ColNames     []string             // Provides names and order of columns
+	ColDefs      map[string]ColumnDef // Provides definition of columns (a map for simpler/faster lookup during type processing)
+	Pks          []IndexKey
+	Fks          []Foreignkey
+	Indexes      []CreateIndex
+	Parent       string //if not empty, this table will be interleaved
+	Comment      string
+	Id           int
+	PrimaryKeyId int
```

Comment:
Instead of primary key id we should have an id per index

Reply:
@shreyakhajanchi PrimaryKeyId int is suggesting by Aditya for cascading effect.

Reply:
@bharadwaj-aditya to comment on this

Reply:
keeping globally unique id's for each schema element will make it very easy to track changes between source and spanner. Also, this will help while implementing cascading effects to elements.

That being said, as PrimaryKeys have a strong 1:1 mapping with the tables, we can evaluate skipping the addition of an extra id here.